core/db/migrate/20150113002122_create_spree_promotion_codes.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSpreePromotionCodes < ActiveRecord::Migration
+  def change
+    create_table :spree_promotion_codes do |t|
+      t.references :promotion, index: true
+      t.string :value
```

Comment:
i know this would be a bit of a change in behavior, but it seems to me that promotion codes should be unique, otherwise we are arbitrarily (db order) finding the first matching code